converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Patient/PatientBirthDateSynchronizer.cs

```diff
@@ -0,0 +1,34 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Dicom;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.Health.DicomCast.Core.Extensions;
+
+namespace Microsoft.Health.DicomCast.Core.Features.Worker.FhirTransaction
+{
+    /// <summary>
+    /// Provides functionality to synchronize DICOM properties to a specific <see cref="Patient.BirthDate"/> property.
+    /// </summary>
+    public class PatientBirthDateSynchronizer : IPatientPropertySynchronizer
+    {
+        /// <inheritdoc/>
+        public void Synchronize(DicomDataset dataset, Patient patient, bool newPatient)
+        {
+            if (newPatient)
+            {
+                EnsureArg.IsNotNull(dataset, nameof(dataset));
+                EnsureArg.IsNotNull(patient, nameof(patient));
+
+                FhirDateTime dicomPatientBirthDate = dataset.GetDatePropertyIfNotDefaultValue(DicomTag.PatientBirthDate);
```

Comment:
nit: FHIR has a Date object type (`Hl7.Fhir.Model.Date`) that only stores the date and you can access throught `BirthDateElement` property. You can avoid making into an object then serialize back into string.